src/extensions/sb-ncbr/partial-charges/color.ts

```diff
@@ -0,0 +1,150 @@
+        [
+            ['atom', 'Atom charges'],
+            ['residue', 'Residue charges'],
+        ],
+        { isHidden: false }
+    ),
+};
+export type PartialChargesThemeParams = typeof PartialChargesThemeParams;
+
+export function getPartialChargesThemeParams() {
+    return PD.clone(PartialChargesThemeParams);
+}
+
+export function PartialChargesColorTheme(
+    ctx: ThemeDataContext,
+    props: PD.Values<PartialChargesThemeParams>
+): ColorTheme<PartialChargesThemeParams> {
+    const model = ctx.structure?.models[0];
+    if (!model) {
+        throw new Error('No model found');
+    }
+    const data = SbNcbrPartialChargesPropertyProvider.get(model).value;
+    if (!data) {
+        throw new Error('No partial charges data found');
+    }
+
+    const { absolute, chargeType } = props;
+    const { typeIdToAtomIdToCharge, typeIdToResidueToCharge, maxAbsoluteAtomCharges, maxAbsoluteResidueCharges } = data;
+    const typeId = SbNcbrPartialChargesPropertyProvider.props(model).typeId;
+    const atomToCharge = typeIdToAtomIdToCharge.get(typeId);
+    const residueToCharge = typeIdToResidueToCharge.get(typeId);
+
+    let maxCharge = 0;
+    if (absolute) {
+        maxCharge = props.maxAbsoluteCharge < 0 ? 0 : props.maxAbsoluteCharge;
+    } else if (chargeType === 'atom') {
+        maxCharge = maxAbsoluteAtomCharges.get(typeId) || 0;
+    } else {
+        maxCharge = maxAbsoluteResidueCharges.get(typeId) || 0;
+    }
+
+    // forces coloring updates
+    const contextHash = SbNcbrPartialChargesPropertyProvider.get(model)?.version;
+
+    const chargeMap = chargeType === 'atom' ? atomToCharge : residueToCharge;
+
+    let color: LocationColor;
+
+    if (!chargeMap) {
+        color = (_: Location): Color => Colors.MissingCharge;
+    } else {
+        color = (location: Location): Color => {
+            let id = -1;
+            if (StructureElement.Location.is(location)) {
+                if (Unit.isAtomic(location.unit)) {
+                    id = StructureProperties.atom.id(location);
+                }
+            } else if (Bond.isLocation(location)) {
+                if (Unit.isAtomic(location.aUnit)) {
+                    const l = StructureElement.Location.create(ctx.structure?.root);
+                    l.unit = location.aUnit;
+                    l.element = location.aUnit.elements[location.aIndex];
+                    id = StructureProperties.atom.id(l);
+                }
+            }
+
+            const charge = chargeMap.get(id);
+
+            if (charge === undefined) {
+                console.warn('No charge found for id', id);
+                return Colors.MissingCharge;
+            }
+
+            return Colors.getColor(charge, maxCharge);
+        };
+    }
+
+    return {
+        factory: PartialChargesColorTheme,
+        granularity: 'group',
+        color,
+        props,
+        description: 'Color atoms and residues based on their partial charge.',
+        preferSmoothing: false,
```

Comment:
Hey @MergunFrimen, could we set `preferSmoothing: true`? Or was there a specific reason to set it to false? It looks much nicer set to true for surfaces by default in my opinion. (Set aside that it is debatable to even use partial charges to color surfaces directly)